src/shims/unix/fs.rs

```diff
@@ -49,6 +49,48 @@ impl FileDescription for FileHandle {
         Ok(self.file.write(bytes))
     }
 
+    fn pread<'tcx>(
+        &mut self,
+        communicate_allowed: bool,
+        bytes: &mut [u8],
+        offset: u64,
+        _ecx: &mut MiriInterpCx<'tcx>,
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        assert!(communicate_allowed, "isolation should have prevented even opening a file");
+        // Emulates pread using seek + read + seek to restore cursor position.
+        // Correctness of this emulation relies on sequential nature of Miri execution.
+        let mut f = || {
+            let cursor_pos = self.file.stream_position()?;
+            self.file.seek(SeekFrom::Start(offset))?;
+            let res = self.file.read(bytes);
+            // Attempt to restore cursor position even if read has failed
+            self.file.seek(SeekFrom::Start(cursor_pos))?;
+            res
+        };
+        Ok(f())
```

Comment:
Why is this wrapped in a closure?

Reply:
To allow use of `?`. Ideally, I would use a `try` block here.

Reply:
Ah, those `?` are for `io::Result`, not `InterpResult`. That's worth a comment.